presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestJdbcRecordSetProvider.java

```diff
@@ -179,13 +179,10 @@ public void testTupleDomain()
 
     private RecordCursor getCursor(JdbcTableHandle jdbcTableHandle, List<JdbcColumnHandle> columns, TupleDomain<ColumnHandle> domain)
     {
-        jdbcTableHandle = new JdbcTableHandle(
-                jdbcTableHandle.getSchemaTableName(),
-                jdbcTableHandle.getCatalogName(),
-                jdbcTableHandle.getSchemaName(),
-                jdbcTableHandle.getTableName(),
-                domain,
-                OptionalLong.empty());
+        jdbcTableHandle = JdbcTableHandle.builder(jdbcTableHandle)
+                .setConstraint(domain)
+                .setLimit(OptionalLong.empty())
```

Comment:
is this code's intention to "strip limit" or it was just doing this to "add domain"?
my guess the latter.
Maybe instead of `.setLimt` add a `checkArgument` asserting that handle doesn't have limit.
(otherwise setting domain would not be correct)